main.py

```diff
@@ -149,15 +211,17 @@ async def send_wrs(self, scores_json, saved_state):
                         msg = await channel.send(score_text)
                         saved_state["recent_scores"].append({**score, "message_id": msg.id})
                     
-                    if rank == 1:
+                    if rank == 1 and msg:
                         saved_state["video_queue"].append({**score, "message_id": msg.id})
+                    # Logically speaking, msg should always be bound, but if it ends up being unbound, log it.
+                    elif not msg:
+                        log("Error: A previous message for a score was not found, and no new message was made!")
```

Comment:
Well if logically speaking it always works, then it's an issue with your tools that they can't tell.
(Even without the if, we'd still get an exception. Which I think is acceptable for something that should logically never happen)

Reply:
While reading the code looks like we'll have 100% coverage, doesn't mean that we have 100% coverage. The only way we can confirm that `msg` is always bound is if we have a code coverage tool that can confirm that with all covered cases, `msg` is bound. 

If that ends up being the case, then we can simplify this code and then use an ignore and make a comment stating that we can confirm `msg` will always be bound.

Reply:
Even then, technically if there really was a way for msg to be None. I'd be fine with it just throwing rather than needing this extra code.

Reply:
Okay, if you insist, I'll replace the much more lenient logging with the stricter exception. I'll use an assert to pull this off.

Reply:
I'm not gonna use an ignore just because it's gonna allow for sloppier code. Ignores should be used sparingly, and there needs to be a good reason to use one. So, if you are able to use a code coverage tool to confirm that `msg` will always be bound in all cases, then we'll have a sufficient reason to use one. 

The only ignore that is used throughout this pull request is with a discord.py decorator, and I use the ignore because the decorator is being used correctly, despite what my type checker is saying. 

Still, I highly suggest that we use an assert statement to ensure `msg` is not `None`. It does what you are looking for and it allows us to only use 1 line of code. It's not that big of a deal.

Reply:
I'm not a fan of it, but sure. Add an assert.